src/main.py

```diff
@@ -38,7 +38,27 @@ def reproduce(population: List[Deer], params: ModelParameters):
 
 
 def naturalDeath(population: List[Deer], params: ModelParameters):
-    pass
+    survivors = []
+    for deer in population:
+        # Determine mortality rate based on the age of the deer
+        if deer.age < 1:
+            mortality_rate = params.calfMortalityRate
+        elif 1 <= deer.age < 5:
+            mortality_rate = params.youngMortalityRate
+        elif 5 <= deer.age < 12:
+            mortality_rate = params.matureMortalityRate
+        else:
+            mortality_rate = params.oldMortalityRate
+        
+        # Determine if the deer survives based on mortality rate
+        if random() > mortality_rate:
+            survivors.append(deer)
+    
+    return survivors
```

Comment:
<img width="602" alt="Screenshot 2024-11-07 at 12 30 36" src="https://github.com/user-attachments/assets/9bb7040c-7262-403f-801d-367c4958fbbb">
This function should be based on the death function from the code